### DISCUSSION TITLE
How to bind command in templated control?

### DISCUSSION THREAD

**Discussion by kochkaqqq** [AvaloniaUI/Avalonia#14378] [Q&A] [answered]
I have Label with PointerPressed event in template.axaml:
`<Label PointerPressed="Label_PointerPressed_1" Content="Label Content" />`
And code behind:
`private void Label_PointerPressed_1(object? sender, Avalonia.Input.PointerPressedEventArgs e)
{
    throw new System.Exception();
}`
But a got the error:
AVLN:0004	Unable to find suitable setter or adder for property PointerPressed of type Avalonia.Base:Avalonia.Input.InputElement for argument System.Runtime:System.String, available setter parameter lists are:
System.EventHandler`1<Avalonia.Input.PointerPressedEventArgs>

**Comment by robloo** [comment, marked as answer]
You aren't binding to a command, you are connecting an event on PointerPressed.

With both events and commands in a templated control remember you can't just connect things with code behind like a UserControl or other view.

1. With Commands
   - Define the command in code-behind, ideally using MVVM toolkit or something `[RelayCommand]`
   - Bind in the XAML control template using a full binding to the templated parent
      - `Command="{Binding PasteCommand, RelativeSource={RelativeSource TemplatedParent}}">`

2. With events
   - Override `OnApplyTemplate(TemplateAppliedEventArgs e)` within the control code-behind and grab a reference to the Label you want (by name). Now you can connect events as you wish.
      - `var myLabel = e.NameScope.Find<Control>("MyLabel");`
      - `myLabel += Label_PointerPressed_1`
   - Keep in mind connecting/disconnecting events in a templated control should be done in OnLoaded and OnUnloaded to avoid other potential concerns
